step-controller/step-controller-server/src/main/java/step/core/scheduler/Executor.java

```diff
@@ -127,12 +142,45 @@ public boolean schedule(ExecutiontTaskParameters task) {
 			logger.error("An error occurred while checking if task exists in scheduler: " + task);
 			throw new RuntimeException(e);
 		}
-		Trigger trigger = TriggerBuilder.newTrigger().withSchedule(CronScheduleBuilder.cronSchedule(task.getCronExpression())).build();
+		//Base calendar allows you to chain calendars.
+		BaseCalendar baseCalendar = new BaseCalendar();
+		if (task.getCronExclusions() != null) {
+			try {
+				for(CronExclusion c : task.getCronExclusions()) {
+					baseCalendar = new CronCalendar(baseCalendar, c.getCronExpression());
+				}
+			} catch (ParseException e) { //such exception should already be caught by validate
+				logAndThrow("One of the cron expressions for the task '" + task.getAttribute(AbstractOrganizableObject.NAME) +
+						"' is invalid.", e);
+			}
+		}
+		//In case exclusions were removed we need to add/replace at least the base calendar
+		try {
+			scheduler.addCalendar("exclusionsCalendar_" + task.getId().toHexString(), baseCalendar, true, false);
+		} catch (SchedulerException e) {//exception throw when adding exclustion calendar
+			logAndThrow("Adding exclusion CRON expressions raised an error for the task '" +
+				task.getAttribute(AbstractOrganizableObject.NAME) + "'.", e);
+		}
+
+		Trigger trigger = TriggerBuilder.newTrigger()
+				.withSchedule(CronScheduleBuilder.cronSchedule(task.getCronExpression()))
+				.modifiedByCalendar("exclusionsCalendar_" + task.getId().toHexString())

```

Comment:
Use a constant for "exclusionsCalendar_" and a function for creating the exclusion calendar id